tests/conftest.py

```diff
@@ -192,158 +209,110 @@ def source_date_epoch(self) -> str:
+    monkeypatch.chdir(tmp_path_factory.getbasetemp())
+    return packages
 
 
 @pytest.fixture
-def package_dynamic_metadata(
-    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+def package_simple_pyproject_ext(
+    tmp_path_factory: pytest.TempPathFactory,
+    monkeypatch: pytest.MonkeyPatch,
 ) -> PackageInfo:
-    package = PackageInfo("dynamic_metadata")
-    process_package(package, tmp_path, monkeypatch)
-    return package
-
-
-@pytest.fixture
-def package_hatchling(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> PackageInfo:
-    package = PackageInfo("hatchling")
-    process_package(package, tmp_path, monkeypatch)
-    return package
-
-
-@pytest.fixture
-def package_simplest_c(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> PackageInfo:
-    package = PackageInfo(
-        "simplest_c",
-    )
-    process_package(package, tmp_path, monkeypatch)
-    return package
-
-
-@pytest.fixture
-def navigate_editable(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> PackageInfo:
-    package = PackageInfo(
-        "navigate_editable",
-    )
-    process_package(package, tmp_path, monkeypatch)
-    return package
-
-
-@pytest.fixture
-def broken_fallback(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> PackageInfo:
     package = PackageInfo(
-        "broken_fallback",
+        "simple_pyproject_ext",
+        tmp_path_factory.mktemp("pkg"),
+        "71b4e95854ef8d04886758d24d18fe55ebe63648310acf58c7423387cca73508",
+        "ed930179fbf5adc2e71a64a6f9686c61fdcce477c85bc94dd51598641be886a7",
+        "0178462b64b4eb9c41ae70eb413a9cc111c340e431b240af1b218fe81b0c2ecb",
+        "de79895a9d5c2112257715214ab419d3635e841716655e8a55390e5d52445819",
     )
-    process_package(package, tmp_path, monkeypatch)
+    process_package(package, monkeypatch)
     return package
 
 
-@pytest.fixture
-def package_sdist_config(
-    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
-) -> PackageInfo:
-    package = PackageInfo(
-        "sdist_config",
+@dataclasses.dataclass(frozen=True)
+class Isolate:
+    state: bool
+    flags: list[str]
+
+
+@pytest.fixture(params=[True, False], ids=["isolated", "not_isolated"])
+def isolate(request: pytest.FixtureRequest, isolated: VEnv) -> Isolate:
+    isolate_request = request.param
+    assert isinstance(isolate_request, bool)
+    if not isolate_request:
+        isolated.prepare_no_build_isolation()
+    flags = []
+    if not isolate_request:
+        flags.append("--no-build-isolation")
+    return Isolate(
+        state=isolate_request,
+        flags=flags,
     )
-    process_package(package, tmp_path, monkeypatch)
-    return package
 
 
-@pytest.fixture
-def package_simple_purelib_package(
-    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
-) -> PackageInfo:
-    package = PackageInfo(
-        "simple_purelib_package",
-    )
-    process_package(package, tmp_path, monkeypatch)
-    return package
-
+@dataclasses.dataclass(frozen=True)
+class Editable:
+    mode: str | None
```

Comment:
Can we at least provide a hint of expected values with `typing.Literal`? Or a comment with a URL or something?

Reply:
Yeah sure thing :+1: